lib/ex_doc/language/erlang.ex

```diff
@@ -215,6 +215,24 @@ defmodule ExDoc.Language.Erlang do
     binary
   end
 
+  defp walk_doc({:code, attrs, [{:a, a_attrs, [a_inner], _a_meta} | _] = a, meta}, _config) do
+    {arity, extra} =
+      case :lists.keydelete(:a, 1, a) do
+        [] ->
+          {0, ""}
+
+        [extra] ->
+          # a_meta is not broken down, so we have to parse stuff like (X, Y)
+          {extra |> String.split(",") |> Enum.count(), extra}
+      end
+
+    type = type_from_fragment(:otp, a_attrs[:href])
```

Comment:
We split "type-" from the actual type. I couldn't find this explicitly done in the current code, but could find composing into "type-<Type>".